pimd/pim_iface.h

```diff
@@ -55,18 +55,25 @@ struct pim_secondary_addr {
 	enum pim_secondary_addr_flags flags;
 };
 
+enum pim_iface_mode {
```

Comment:
is it ever going to be possible to have a interface act as both SPARSE and DENSE or is everything mutually exclusive?

Reply:
yes, it is possible, that is sparse-dense mode. The common config/use-case is to use sparse mode for groups that have RP mapping, otherwise fallback to dense mode.

Reply:
then shouldn't these be flags that we can set?  An enum doesn't cut it or we need to add a both mode to the enum

Reply:
Yeah, that  is what `SPRSE_DENSE` is mode is. This in part reflects that fact that other vendors have the mixed mode of `sm` and `dm` as its own mode/config, not two independent ones.  I.e, if you want o configure that mode you don't do 
```
interface eth0
  ip pim sm
  ip pim dm
```

instead you do 
```
interface pim sm-dm
```
That is makes it cleat you are running in mixed mode rather having two separate lines. 

Of course this can still be done using flags, implementation details. Since the config is separate, I treated the code the same way and went with enums and I don't have a a strong feeling either way.